### PR TITLE
VirtualNetworkGateway: Enabling connection on priavte ip.

### PR DESCRIPTION
**Pull request by nalinrajgupta**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json [hunk 1]
```diff
@@ -2898,6 +2907,10 @@
           "type": "boolean",
           "description": "EnableBgp flag."
         },
+        "useLocalAzureIpAddress": {
+          "type": "boolean",
+          "description": "Use private local azure IP for the connection."
```
Comment (anthony-c-martin) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json]: [nit] Capitalize Azure

File: specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json [hunk 2]
```diff
@@ -2898,6 +2907,10 @@
           "type": "boolean",
           "description": "EnableBgp flag."
         },
+        "useLocalAzureIpAddress": {
```
Comment (anthony-c-martin) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json]: `useLocalAzureIpAddress` and `enablePrivateIpAddress` sound very similar to me and could be confusing to customers. Are there any more descriptive names that could help differentiate these two settings?

**Review by anthony-c-martin — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json (hunk 2 above)
Comment (nalinrajgupta) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json]: These setting are on two different resources, and makes sense accordingly.

Comment (anthony-c-martin) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/virtualNetworkGateway.json]: Ah, I missed that. Thanks for clarifying!

**Comment by allenjzhang** [comment]
Please review validation failure and ensure checks are passing. Thx

**Comment by number213** [comment]
NetowrkValdiation failure was because of descriptions in another files that carried over from master. I copied changes from this PR https://github.com/Azure/azure-rest-api-specs/pull/8236

**Review by anton-evseev — APPROVED** [review]

**Pull request merged**